presto-main/src/main/java/io/prestosql/execution/QueryPreparer.java

```diff
@@ -42,17 +44,20 @@
 public class QueryPreparer
 {
     private final SqlParser sqlParser;
+    private Provider<StatementAccessControl> stmtAccessControl;
 
     @Inject
-    public QueryPreparer(SqlParser sqlParser)
+    public QueryPreparer(SqlParser sqlParser, Provider<StatementAccessControl> stmtAccessControl)
     {
         this.sqlParser = requireNonNull(sqlParser, "sqlParser is null");
+        this.stmtAccessControl = stmtAccessControl;
```

Comment:
`stmtAccessControl` should be non-null:
```
this.stmtAccessControl = requireNonNull(stmtAccessControl, "stmtAccessControl is null");
```

Reply:
ok. Specifically left it to act based on review comments.